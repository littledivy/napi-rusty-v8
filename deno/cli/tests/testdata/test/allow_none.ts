import { unreachable } from "../../../../test_util/std/testing/asserts.ts";

const permissions: Deno.PermissionName[] = [
  "read",
  "write",
  "net",
  "env",
  "run",
  "ffi",
  "hrtime",
];

for (const name of permissions) {
  Deno.test({
    name,
    permissions: {
      [name]: true,
    },
    fn() {
      unreachable();
    },
  });
}
