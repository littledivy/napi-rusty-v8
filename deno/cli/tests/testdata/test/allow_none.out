[WILDCARD]
running 7 tests from [WILDCARD]
test read ... FAILED [WILDCARD]
test write ... FAILED [WILDCARD]
test net ... FAILED [WILDCARD]
test env ... FAILED [WILDCARD]
test run ... FAILED [WILDCARD]
test ffi ... FAILED [WILDCARD]
test hrtime ... FAILED [WILDCARD]

failures:

read
PermissionDenied: Can't escalate parent thread permissions
[WILDCARD]

write
PermissionDenied: Can't escalate parent thread permissions
[WILDCARD]

net
PermissionDenied: Can't escalate parent thread permissions
[WILDCARD]

env
PermissionDenied: Can't escalate parent thread permissions
[WILDCARD]

run
PermissionDenied: Can't escalate parent thread permissions
[WILDCARD]

ffi
PermissionDenied: Can't escalate parent thread permissions
[WILDCARD]

hrtime
PermissionDenied: Can't escalate parent thread permissions
[WILDCARD]

failures:

	read
	write
	net
	env
	run
	ffi
	hrtime

test result: FAILED. 0 passed; 7 failed; 0 ignored; 0 measured; 0 filtered out [WILDCARD]
